backend/models/node.py

```diff
@@ -1,87 +1,249 @@
-from .shared import db
-from .measurement import Measurement
+from .shared import db # Allows the models to be split out into separate files.
 
 
 class Node(db.Model):
-    __tablename__ = "node"
+    """
+    A class used to generate a table of Nodes.
+    Nodes are used to model Hierarchy trees that consist of objectives,
+    sub-objectives, and terminating in measurements.
+
+    Based off of SQLALchemy's adjacency_list.py:
+    https://docs.sqlalchemy.org/en/14/_modules/examples/adjacency_list/adjacency_list.html
+
+    ...
+
+    Attributes
+    ----------
+    id : int
+        Node's unique id.
+    parent_id : int
+        Parent node's unique id.
+    hierarchy_id : int
+        Id of the hierarchy the node belongs to.
+    name : str
+        The name of the node.
+    icon : str
+        String to store the icon the frontend uses for the node.
+    weight : float
+        Floating point value representing weight
+    is_measurement : bool
+        Flag for frontend.
+    measurement_type : str
```

Comment:
measurement_type and value_function seem like they should be grouped together as a measurement object since both are only necessary to a measurement node. It'd also be easy to check if a node was a measurement node since measurement would not equal null. It wouldn't require any extra endpoints since you'd just be working with a node.

Reply:
Adjusted to_dict() to return a measurement object containing both fields.

value_function has been renamed to valueFunction in the dict for consistency